### DISCUSSION TITLE
Can we have the ability to reverse the flow of a Sparkline so newest is on the left as opposed to the right

### DISCUSSION THREAD

**Discussion by rhacer63** [Textualize/textual#3987] [Ideas]
Sparklines flow right to left, would it be possible to add an option to get them to flow left to right?

**Comment by davep** [comment]
What would be the utility of this over reversing the data provided to the widget?

```python
from textual.app import App, ComposeResult
from textual.widgets import Sparkline

class ReversedSparklineApp(App[None]):

    DATA: list[float] = list(range(20))

    def compose(self) -> ComposeResult:
        yield Sparkline(self.DATA)
        yield Sparkline(list(reversed(self.DATA)))


if __name__ == "__main__":
    ReversedSparklineApp().run()
```

**Comment by rhacer63** [comment]
@davep

That's a good question, I have time-series data that fills in over time and I could reverse it, but it would still fill in right to left over time which while not a crisis would look a bit counter-intuitive.

Image below shows the data after about five polls.

![image](https://github.com/Textualize/textual/assets/27306556/a648ba9f-f77a-4cc5-a2f7-839e40a4f98c)

As you can see the left is empty, if I reversed the data, the left would still be empty.

**Reply by davep** [reply]
I think I might be misunderstanding this somewhat. A `Sparkline` doesn't "fill" from either end, it shows all of the available data within the space of the `Sparkline`. For example:

```python
from textual.app import App, ComposeResult
from textual.widgets import Sparkline

class SparklineApp(App[None]):

    def compose(self) -> ComposeResult:
        for n in range(40):
            yield Sparkline(list(range(n+1)))


if __name__ == "__main__":
    SparklineApp().run()
```

giving:

![Screenshot 2024-01-09 at 14 38 46](https://github.com/Textualize/textual/assets/28237/e24e05f6-e737-4ab4-a723-de54bbfba560)

the reversed version:

```python
from textual.app import App, ComposeResult
from textual.widgets import Sparkline

class SparklineApp(App[None]):

    def compose(self) -> ComposeResult:
        for n in range(40):
            yield Sparkline(list(reversed(range(n+1))))


if __name__ == "__main__":
    SparklineApp().run()
```

giving:

![Screenshot 2024-01-09 at 14 51 50](https://github.com/Textualize/textual/assets/28237/0ada2d68-a554-43d3-af78-1bde40c96665)

Do you think you could show a MRE of what you're doing to help better understand what you're looking for?

**Comment by rhacer63** [comment]
@davep 

In sorting out that MRE, I think I got things sorted out. Thank you for your help.

**Reply by davep** [reply]
Glad to hear you got it sorted.
